models/user.js

```diff
@@ -20,10 +22,28 @@ module.exports = (sequelize, DataTypes) => {
       type: DataTypes.BOOLEAN,
       defaultValue: false
     },
-  });
+  }, {
+    hooks: {
+      beforeCreate: async (user) => {
+        await user.hashPassword();
+      }
+    }
+  }
+  );
+
   User.associate = models => User.hasOne(models.Profile, {
     foreignKey: 'userId',
     onDelete: 'CASCADE'
   });
+
+  User.prototype.hashPassword = async function hashPassword() {
+    const saltRounds = 10;
+    this.password = await bcrypt.hash(this.password, saltRounds);
```

Comment:
Good job.

Just return `this.password` here after hashing, and then we are set.

Reply:
Alright